### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by kponichtera**

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+RUN curl -o task.deb -L ${TASK_URL} \
+    && dpkg -i task.deb \
+    && rm task.deb
+
+WORKDIR /build
+
+# Warm up the cache
+ENV npm_config_cache /npm-cache
+COPY package.json package-lock.json ./
+RUN npm install \
```
Comment (kbaran1998) [inline frontend/Dockerfile]: I would recommend using `npm ci` for production. It is more deterministic as it installs the exact versions from `package.json` and `package-lock.json` and only performs read operations on them, unlike `npm install`.

File: frontend/src/app/components/inference-form/components/tags-input/tags-input.component.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  separatorKeysCodes: number[] = [ENTER, COMMA];
+
+  private _newTag: string = ''
+
+  @Input()
+  disabled: boolean = false;
+
+  @Input()
+  tags: string[] = []
```
Comment (kbaran1998) [inline frontend/src/app/components/inference-form/components/tags-input/tags-input.component.ts]: We could use a set instead. Tags do not repeat.

File: interface_service/src/main.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 """Main file for the FastAPI application."""
 from typing import Set
 from fastapi import FastAPI
+from fastapi.responses import JSONResponse
```
Comment (kbaran1998) [inline interface_service/src/main.py]: Is this necessary to import it if it is not used?

**Review by kbaran1998 — CHANGES_REQUESTED** [review]

File: frontend/Dockerfile (hunk 1 above)
Comment (kponichtera) [reply frontend/Dockerfile]: I didn't know about that one, I'll replace `install` with `ci` throughout the project and test if everything works.

File: interface_service/src/main.py (hunk 3 above)
Comment (kponichtera) [reply interface_service/src/main.py]: It's not, looks like a leftover from when I was playing around with FastAPI's JSON serialization.

File: frontend/src/app/components/inference-form/components/tags-input/tags-input.component.ts (hunk 2 above)
Comment (kponichtera) [reply frontend/src/app/components/inference-form/components/tags-input/tags-input.component.ts]: I've refactored it into `Set<string>`.

**Review by kbaran1998 — APPROVED** [review]
LGTM

**Pull request merged**